cleveref-usedon.dtx

```diff
@@ -493,35 +533,33 @@
 %
 %   \subsection{Options and requirements}
 %
-%	The following package is included in the \LaTeX{} kernel since 2020-10-01.
-%   Here, it is manually |\Require|'d for users with older \LaTeX{} versions.
-%   For such users the loading should have been aborted anyway.
+%	Below are the available package options.
 %    \begin{macrocode}
-\RequirePackage{xparse}
-%    \end{macrocode}
-%
-%	The following package options currently don't do anything.
-%    \begin{macrocode}
-\bool_new:N \g_@@_StandardBehaviour_bool
-\bool_gset_true:N \g_@@_StandardBehaviour_bool
-\DeclareOption{usedon}{
-    \OptionNotUsed
-    \bool_gset_true:N \g_@@_StandardBehaviour_bool
-}
-\DeclareOption{notusedon}{
-    \OptionNotUsed
-    \bool_gset_false:N \g_@@_StandardBehaviour_bool
-}
+
+\keys_define:nn { cleveref-usedon }
+  {
+    , Default     .tl_set:N   = \l_@@_default_tl
```

Comment:
1. Actually this was learned from others, by putting commas in the front it becomes easier to make the code aligned. An empty key would simply be ignored.
2. This was also learned from others, many people told me to reduce the use of global variables unless really necessary (this was during the development of `crefthe`, which was my first package in expl3 language, at the time I know very less about how to write code with it). That is why I usually use local variables for handling options.

Reply:
I actually have not yet fully read up on and understood how exactly scoping works in LaTeX3. 
Your expertise is definitely much more extensive than mine :)
Nevertheless, I think this makes the code further down more readable: 
```text
\l_@@_default_package_option_tl
```

Reply:
Sure, I shall change this variable name, along with many other of your suggestions.

I am kind of thinking to integrate the code in #12 before we change the variable names, this way one can easily do a find-and-replace afterwards. But I feel unsure how to do so. If I push any new commits, it would certainly interfere with the current state, and I am afraid that your comments would be missing.

The code was written two or three weeks ago, so I also need some time to review. However, my new semester shall begin tomorrow, so I may not have very much time on this in the next two months, but I shall try to modify the code gradually.

Reply:
I propose you make a final commit to this PR which includes your changes to my review comments and afterwards I'll merge it to `main`. 

I assume you have another local branch for your upcoming features? 
If so, you can do a `git pull origin main` to pull the most recent version of `main` into your local branch -- you will probably get a few merge conflicts that you have to resolve locally. Finally, you can make a new PR with those resolved conflict + your new features.

Reply:
Thank you! Does this mean that I can collect several commits on my local side, and after I have confirmed that all the issue you mentioned in the comments have been corrected, I can then push the final result to my fork of this repository, and then you can check whether it is OK to be merged. Is this correct?

Sorry for my poor knowledge on Git. I have actually been using the GUI application of GitHub, which is very simple, each time I just write a summary on the changes and then click the "Commit" button, but I don't know other features anymore :(  As you might have noticed, this "fork and pull-request" is my first time trying to do something other than "commit".

Reply:
> Nevertheless, I think this makes the code further down more readable:
> 
> ```
> \l_@@_default_package_option_tl
> ```

Done (2023-11-10).